### PR TITLE
PrivacyInfo Files for PPCP Modules

### PR DESCRIPTION
**Pull request by KunJeongPark**
### Summary of changes

- Create PrivacyInfo.xcprivacy files for all PPCP modules -  CorePayments, CardPayments, PayPalNativePayments,
PaymentButtons, PayPalWebPayments and FraudProtection 

Generated Privacy Report from demo app:
[Demo-PrivacyReport 2024-03-08 07-52-14_4.pdf](https://github.com/paypal/paypal-ios/files/14540209/Demo-PrivacyReport.2024-03-08.07-52-14_4.pdf)

### Checklist

- [x] Added a changelog entry

- @KunJeongPark

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,11 +17,19 @@
...
   * Add `PayPalVaultResult` type to return vault result
   * Add `PayPalVaultDelegate` to handle results from vault flow
   * Add `PayPalWebCheckoutClientError.paypalVaultResponseError` for missing or invalid response from vaulting
+  * Add PrivacyInfo.xcprivacy file
+* CorePayments
+  * Add PrivacyInfo.xcprivacy file
+* PayPalNativePayments
+  * Add PrivacyInfo.xcprivacy file
+* FraudProtection
+  * Add PrivacyInfo.xcprivacy file
```
Comment (scannillo) [inline CHANGELOG.md]: Take it or leave it comment - IMO one changelog entry for all the modules is fine for the PrivacyManifest additions!

**Comment by scannillo** [comment]
This PR is awesome! 👏 Thanks also for including the PDF

Looking at the PDF you uploaded, it's a odd that Core & Buttons trigger an entry for "Additional Data". Poking around it looks like if we make the file something like:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>NSPrivacyCollectedDataTypes</key>
	<array>
	</array>
</dict>
</plist>
```

it doesn't trigger that "Additional Data" field in the privacy report

File: CHANGELOG.md (hunk 1 above)
Comment (jaxdesmarais) [reply CHANGELOG.md]: I think previously for Apple mandated things we've also linked to an Apple post with the requirements, so we can also consider adding something like [meets Apple's new Privacy Update requirements](https://developer.apple.com/news/?id=3d8a9yyh) - the same syntax should work in the changelog for links as here, so can do: `[my very cool text here](www.myurl.com)`

File: PayPal.podspec [hunk 2]
```diff
@@ -14,12 +14,14 @@ Pod::Spec.new do |s|
   s.subspec "CardPayments" do |s|
     s.source_files  = "Sources/CardPayments/**/*.swift"
     s.dependency "PayPal/CorePayments"
+    s.resource_bundle = { "PayPalCard_PrivacyInfo" => "Sources/CardPayments/PrivacyInfo.xcprivacy" }
```
Comment (jaxdesmarais) [inline PayPal.podspec]: Looks like this one should be `CardPayments`:
```suggestion
    s.resource_bundle = { "CardPayments_PrivacyInfo" => "Sources/CardPayments/PrivacyInfo.xcprivacy" }
```

File: PayPal.podspec [hunk 3]
```diff
@@ -14,12 +14,14 @@ Pod::Spec.new do |s|
...
   end
 
   s.subspec "PayPalNativePayments" do |s|
    s.source_files  = "Sources/PayPalNativePayments/**/*.swift"
    s.dependency "PayPal/CorePayments"
    s.dependency "PayPalCheckout", "1.2.0"
+   s.resource_bundle = { "PayPalNative_PrivacyInfo" => "Sources/PayPalNativePayments/PrivacyInfo.xcprivacy" }
```
Comment (jaxdesmarais) [inline PayPal.podspec]:
```suggestion
   s.resource_bundle = { "PayPalNativePayments_PrivacyInfo" => "Sources/PayPalNativePayments/PrivacyInfo.xcprivacy" }
```

File: PayPal.podspec [hunk 4]
```diff
@@ -28,11 +30,13 @@ Pod::Spec.new do |s|
     s.resource_bundle = {
     	'PayPalSDK' => ['Sources/PaymentButtons/*.xcassets']
     }
+    s.resource_bundle = { "PayPalButtons_PrivacyInfo" => "Sources/PaymentButtons/PrivacyInfo.xcprivacy" }
```
Comment (jaxdesmarais) [inline PayPal.podspec]:
```suggestion
    s.resource_bundle = { "PaymentButtons_PrivacyInfo" => "Sources/PaymentButtons/PrivacyInfo.xcprivacy" }
```

File: PayPal.podspec [hunk 5]
```diff
@@ -28,11 +30,13 @@ Pod::Spec.new do |s|
...
   end
 
   s.subspec "PayPalWebPayments" do |s|
     s.source_files  = "Sources/PayPalWebPayments/*.swift"
     s.dependency "PayPal/CorePayments"
+    s.resource_bundle = { "PayPalWeb_PrivacyInfo" => "Sources/PayPalWebPayments/PrivacyInfo.xcprivacy" }
```
Comment (jaxdesmarais) [inline PayPal.podspec]:
```suggestion
    s.resource_bundle = { "PayPalWebPayments_PrivacyInfo" => "Sources/PayPalWebPayments/PrivacyInfo.xcprivacy" }
```

**Review by jaxdesmarais — COMMENTED** [review]
It looks like we are missing an entry in the `podspec` for `CorePayments` and `FraudProtection`

File: PayPal.podspec (hunk 2 above)
Comment (KunJeongPark) [reply PayPal.podspec]: Oh great catch about the CorePayments and Cards. This one,  I just prepended PayPal as opposed to Braintree for all the names. But I agree it's confusing

Comment (jaxdesmarais) [reply PayPal.podspec]: Ah - yeah those are the actual module names on the BT side: https://github.com/braintree/braintree_ios/tree/main/Sources where PPCP doesn't use the prefix in the module names: https://github.com/paypal/paypal-ios/tree/main/Sources. I think we can follow a similar pattern where we use the module names from sources on both.

**Review by jaxdesmarais — APPROVED** [review]
👏

**Review by richherrera — APPROVED** [review]
💯 !!!

**Pull request merged**